authorize_error.go

```diff
@@ -49,17 +49,21 @@ func (f *Fosite) WriteAuthorizeError(rw http.ResponseWriter, ar AuthorizeRequest
 	}
 
 	redirectURI := ar.GetRedirectURI()
+	error_description := rfcerr.Description
 	query := url.Values{}
 	query.Add("error", rfcerr.Name)
-	query.Add("error_description", rfcerr.Description)
 	query.Add("state", ar.GetState())
-	if f.SendDebugMessagesToClients && rfcerr.Debug != "" {
-		query.Add("error_debug", rfcerr.Debug)
-	}
-
+	// We expose both error hint and debug strings through standard error description, too
+	// (they are non-standard fields and some clients do not show them).
 	if rfcerr.Hint != "" {
 		query.Add("error_hint", rfcerr.Hint)
+		error_description += ": " + rfcerr.Hint
+	}
+	if f.SendDebugMessagesToClients && rfcerr.Debug != "" {
+		query.Add("error_debug", rfcerr.Debug)
+		error_description += " (" + rfcerr.Debug + ")"
```

Comment:
We could make this also more readable by using newlines:

```
${description}

${description}

Debug log: ${debug}
```

Reply:
Sure.